unit_test.cpp

```diff
@@ -0,0 +1,121 @@
+#include "Alien.h"
+#include "BaseballPlayer.h"
+#include "BaseballMitt.h"
+#include <unistd.h>
+#include <thread>
+#include <memory>
+#include <mutex>
+#include <iostream>
+#include <assert.h>
+
+#undef NDEBUG
+
+#ifndef NDEBUG
+#   define ASSERT(condition, message) \
+    do { \
+        if (! (condition)) { \
+            std::cerr << "Assertion `" #condition "` failed in " << __FILE__ \
+                      << " line " << __LINE__ << ": " << message << std::endl; \
+            std::terminate(); \
```

Comment:
Could `throw std::runtime_error();`

Reply:
Ahh ok if i wanted to instead catch it and move on with other tests then report at the end. I think this is fine for now but that is a nice way of doing it.